Running on gaea

rgrumbine opened the pull request:
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>


## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    Macros and env to run on gaea
- [x] Developer(s): 
    Robert Grumbine
- [x] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    executes quick_suite
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No

DeniseWorthen commented:
Is there a reason not to add C6 in addition to C5?

DeniseWorthen commented on configuration/scripts/machines/Macros.gaea_cray:
```diff
@@ -0,0 +1,57 @@
+#==============================================================================
+# Makefile macros for NOAA hera, intel compiler
```
hera->gaeaC5

DeniseWorthen commented on configuration/scripts/machines/env.gaea_intel:
```diff
@@ -7,20 +7,27 @@ endif
 
```
The env.gaea should probably be env.gaeac5 since this I presume gaeac6 would be different.

apcraig commented:
I'm happy to merge this when ready.  @rgrumbine @DeniseWorthen, just let me know when you're happy and when it's ready.

DeniseWorthen commented:
@rgrumbine Gaea C5 and C6 are different machines. If the intent is to enable on Gaea, it needs to be clear which gaea.

rgrumbine commented:
Agreed Denise. I'll make the update(s)

rgrumbine commented:
I've made the changes and passed the quick_start suite.

DeniseWorthen commented:
Are files ``Macros.gaea_intel`` and ``env.gaea_intel`` still required?

rgrumbine commented on configuration/scripts/machines/env.gaea_cray:
```diff
@@ -0,0 +1,44 @@
+setenv ICE_MACHINE_ENVINFO "ifort 18.0.5 20180823, cray-mpich, cray-netcdf"
+setenv ICE_MACHINE_MAKE gmake
+setenv ICE_MACHINE_WKDIR $HOME/scratch/CICE_RUNS
+setenv ICE_MACHINE_INPUTDATA /ncrc/home1/Robert.Grumbine/rgdev/CICE_INPUTDATA
```
Macros.gaea_intel and env.gaea_intel aren't required, but do work on both c5 and c6.

The path is actually a link to a world-shared disk. Specifying it this way, though, gives some parallelism between different platforms. For gaea, hera, and wcoss2, the path is $BOBHOME/rgdev/CICE_INPUTDATA

DeniseWorthen replied:
OK, but Gaea provides "world-shared" project space for this purpose, eg ``/gpfs/f5/epic/world-shared/...``

rgrumbine commented:
I agree about avoiding over-specific file paths. In this case, it is as for using Tony Craig's path on hera and other systems -- to have a shared location for the CICE input data files. At least when I started, Tony wasn't on gaea.

@DeniseWorthen I think I addressed your concerns about c5 vs. c6
